core/shared/src/main/scala-2.12/zio/stm/ZSTMUtils.scala

```diff
@@ -0,0 +1,16 @@
+package zio.stm
+
+import scala.collection.mutable
```

Comment:
in your experience, are scala mutable collections on par with java ones performance-wise?

Reply:
It's extremely case-specific. Usually if you need to convert to/from Scala's immutable collections (which we do a lot in this file), Scala's mutable collections are faster.

Otherwise, I found them to be quite case-specific depending on the nature of the keys. I don't remember exactly under what conditions, but sometimes Java's is faster while other times Scala's is faster